action/execution.go

```diff
@@ -117,7 +117,10 @@ func (ex *Execution) LoadProto(pbAct *iotextypes.Execution) error {
 
 	ex.contract = pbAct.GetContract()
 	ex.amount = &big.Int{}
-	ex.amount.SetString(pbAct.GetAmount(), 10)
+	_, ok := ex.amount.SetString(pbAct.GetAmount(), 10)
```

Comment:
As mentioned in the [issue](https://github.com/iotexproject/iotex-core/issues/3042),  the value of ex.amont will change even if there is an error when doing `SetString()`
Use `x, ok = new(big.Int).SetString(string, 10)` pattern instead